### PR TITLE
Add Section Header component

### LINKED ISSUE #1910: [COMPONENT] Enable use of header in declarative list
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
I would like to be able to use a kirby-item-divider or similar, to declaratively separate lists, groups or items, optionally with some text, acting as a sort of sub-header or footer. 

Together with #1816 it should provide us with the necessary structure to replicate the current look and feel of the list, once we tackle the styling of these in #1814 and #1817. 

## Describe the solution you'd like
I would expect to be able to do something like 
```
<kirby-list>
    <kirby-list-section>
        <kirby-item-divider>
            ...
        </kirby-item-divider>

        <kirby-list-item>
            ...
        </kirby-list-item>
    <kirby-list-section>
</kirby-list>
```

This would mean that Kirby provides a thin wrapper around the `ion-item-divider` that is currently used within the `kirby-list` so it can be provided directly instead.  

## Have you considered any alternatives?
.

## Are there any additional context?
See #1794 and #1813 for context.

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### LINKED ISSUE #1916: [Enhancement] Enable detail, label and title styling for SectionHeader
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
I would like to be able to style the fields of the [Section Header](https://zpl.io/bL3oDyQ) component, so that it is as easy as possible to create the default states of the component in Zeplin.   

## Describe the solution you'd like
A class or attribute (similar to our way of styling paragraphs and headings in Item), that can be added to my paragraphs and headings.

## Are there any additional context?
Once this is possible, we are able to deprecate the existing `kirby-list-section-header`, and replace it with guidance for creating a similar look with the new declarative component: 
```
<kirby-section-header>
    <span class="kirby-text-small">Section</span>
</kirby-section-header>
```

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by RasmusKjeldgaard**
## Which issue does this PR close?

This PR closes #1910
This PR closes #1916 

## What is the new behavior?
A new `SectionHeader` is introduced. This new component replaces the ListSectionHeader component, and should better reflect the [actual design of the section header](https://zpl.io/aB39DjO). 

The reason this is a new component, is that we want to deprecate the old list-section-header, as the component has a wrong name, and is also relevant outside lists.

A showcase is added, and can be found under 'Section Header' in the cookbook. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any additional context?
This issue is a part of #1883 

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be automatically merged to master via [automerge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request).

### CODE REVIEW

File: libs/designsystem/src/lib/components/section-header/_section-header.utils.scss [hunk 1]
```diff
@@ -0,0 +1,33 @@
+@import '~@kirbydesign/core/src/scss/utils';
+
+/*
+* This mixin is applied both in the section-header styles, for using 
+* headings and paragraphs directly, and in kirby-label styles, for 
+* wrapping them in a kirby-label within a section-header. 
+*/
+
+@mixin section-header-typography() {
+  --ion-text-color-rgb: #{get-text-color('black')}; //TODO: should we make this configurable?
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/section-header/_section-header.utils.scss:10]: Consider this.

File: libs/designsystem/src/lib/components/section-header/section-header.component.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+:host {
+  ion-item-divider {
+    --inner-padding-end: 0;
+    --color: unset;
+    --background: transparent;
+    font-weight: inherit;
+    // TODO: Determine if we really want this.
+    // This way, section header cannot be used as simple spacer
+    // Without supplying a piece of content with the needed height
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/section-header/section-header.component.scss:12]: But maybe that is fine. I dont think we should start defining a minimum height larger than 0 here. If a simple spacing is needed, it should not be the responsibility of the section header, but rather a class provided by us, or a custom component in the project.

**Comment by RasmusKjeldgaard** [comment]
@jkaltoft The Section Header has its own cookbook entry in the menu. 
Furthermore, there is a small example where it has been used in the still hidden item-group showcase, which is why there are quite some changes to those showcase/example-files too. It can be found here: https://cookbook.kirby.design/branch/kby-1910-section-header/#/home/showcase/item-group

File: libs/designsystem/src/lib/components/section-header/section-header.component.scss (hunk 2 above)
Comment (jkaltoft) [reply libs/designsystem/src/lib/components/section-header/section-header.component.scss:12]: Agree.

File: libs/designsystem/src/lib/components/section-header/section-header.component.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    min-height: 0;
+
+    @include section-header.typography();
```
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/section-header/section-header.component.scss:15]: Consider moving this up _before_ other "normal" declarations, but _after_ custom property declarations.

**Review by jkaltoft — APPROVED** [review]
Nice solution. I think the showcase examples are good.

Only issue I have is with the markup. Using `<h3>` for different levels is not the best example, but I know we used this markup before, so it doesn't have to be changed in this PR. It's probably best to fix it on its own when we focus on accessibility later.

❗️ Don't forget to remove the TODOs.

**Review by jkaltoft — DISMISSED** [review]
image

**Review by jkaltoft — APPROVED** [review]

**Pull request merged**